packages/web/src/common/store/tipping/selectors.ts

```diff
@@ -162,6 +152,43 @@ export const getOptimisticSupporters = createSelector(
   }
 )
 
+export const getOptimisticSupporting = createSelector(
+  getSupporting,
+  getSupportingOverrides,
+  getOptimisticSupporters,
+  (supporting, supportingOverrides, supportersMap) => {
+    /**
+     * Merge supporting maps
+     */
+    const mergedMap = mergeMaps<SupportingMap>({
+      map: supporting,
+      mapOverrides: supportingOverrides
+    })
+
+    /**
+     * Get updated ranking from optimistic supporters map
+     */
+    const result: SupportingMap = {}
+    const mergedUserIds = Object.keys(mergedMap) as unknown as ID[]
+    for (const userId of mergedUserIds) {
+      result[userId] = {}
+      const supportingUserIds = Object.keys(
+        mergedMap[userId]
+      ) as unknown as ID[]
+      for (const supportingUserId of supportingUserIds) {
+        result[userId][supportingUserId] = {
```

Comment:
I don't think i need this reducer-like immutability here, do I? I should be able to just change rank directly on `mergedMap` right? Testing now...